canvas/animation.go

```diff
@@ -19,47 +19,58 @@ const (
 // The content of fn should apply the color values to an object and refresh it.
 // You should call Start() on the returned animation to start it.
 func NewColorRGBAAnimation(start, stop color.Color, d time.Duration, fn func(color.Color)) *fyne.Animation {
+	r1, g1, b1, a1 := start.RGBA()
+	r2, g2, b2, a2 := stop.RGBA()
+
+	rDelta := int(r2) - int(r1)
+	gDelta := int(g2) - int(g1)
+	bDelta := int(b2) - int(b1)
+	aDelta := int(a2) - int(a1)
```

Comment:
I suggest to store these (and all other deltas) as `float32` because it is used in a `float32` computation.
Colour channels should also be shifted 8 bit to the right before converted into `float32` because this saves the shift in the `scaleChannel` function.

Reply:
Unfortunately I don't think we can pre-shift, because the delta can be a negative number...